src/Urls/UrlRecord.php

```diff
@@ -162,4 +165,19 @@ public static function exists($slug, string $locale = null, Model $ignoredModel
 
         return ($builder->count() > 0);
     }
+
+    public static function allOnlineModels(string $locale): Collection
+    {
+        $records = static::where('locale', $locale)
+            ->where('redirect_id', '=', null)
+            ->orderBy('updated_at', 'DESC')
+            ->get();
+
+        // Filter out offline urls... TODO: this should be a state that is owned by the url and not the model.
+        return $records->map(function(UrlRecord $urlRecord){
+            return Morphables::instance($urlRecord->model_type)->find($urlRecord->model_id);
+        })->reject(function(ProvidesUrl $model){
```

Comment:
@BenCavens In deze pipeline, na de map, zitten null values. Dit omdat de url van gearchiveerde paginas nog wel in de urls table zitten en deze hier niet opgehaald kunnen worden. Een mogelijke oplossing is voor de online reject te doen alle null values uit deze collection halen.
```suggestion
        })->reject(function($model){return $model == null})->reject(function(ProvidesUrl $model){
```

Reply:
lijkt me goed. 👍 
zet er mss nog comment bij dat model null kan zijn indien deze gearchiveerde status hebben.

Reply:
will do!